src/main/java/seedu/address/logic/commands/FindCommand.java

```diff
@@ -19,48 +26,77 @@ public class FindCommand extends Command {
             return new CommandResult(
                     String.format(Messages.MESSAGE_TUTORS_LISTED_OVERVIEW, model.getFilteredTutorList().size()));
-
-        }
-
-        if (model.getCurrentListType() == Model.ListType.PERSON_LIST) {
-            model.updateFilteredPersonList(predicate);
-            return new CommandResult(String.format(Messages.MESSAGE_PERSONS_LISTED_OVERVIEW,
-                    model.getFilteredPersonList().size()));
-
-        }
-
-        if (model.getCurrentListType() == Model.ListType.TUITIONCLASS_LIST) {
-            model.updateFilteredTuitionClassList(predicate);
-            return new CommandResult(String.format(Messages.MESSAGE_TUITIONCLASSES_LISTED_OVERVIEW,
-                    model.getFilteredTuitionClassList().size()));
+        case PERSON_LIST:
+            this.personPredicate = new NameContainsKeywordsPredicate<>(keywords);
+            model.updateFilteredPersonList(personPredicate);
+            return new CommandResult(
+                    String.format(Messages.MESSAGE_PERSONS_LISTED_OVERVIEW, model.getFilteredPersonList().size()));
+        default:
+            assert (type == ListType.TUITIONCLASS_LIST);
+            this.classPredicate = new NameContainsKeywordsPredicate<>(keywords);
+            model.updateFilteredTuitionClassList(classPredicate);
+            return new CommandResult(
+                    String.format(Messages.MESSAGE_TUITIONCLASSES_LISTED_OVERVIEW,
+                            model.getFilteredTuitionClassList().size()));
         }
-        throw new ClassCastException();
     }
 
     @Override
     public boolean equals(Object other) {
-        return other == this // short circuit if same object
-                || (other instanceof FindCommand // instanceof handles nulls
-                && predicate.equals(((FindCommand) other).predicate)); // state check
+        if (other == this) {
+            return true;
+        }
+
+        if (other instanceof FindCommand) {
+            FindCommand otherFind = (FindCommand) other;
+
+            if (this.studentPredicate != null) {
+                if (otherFind.studentPredicate != null) {
+                    return this.studentPredicate.equals(((FindCommand) other).studentPredicate);
+                }
+                return false;
+            } else if (this.tutorPredicate != null) {
+                if (otherFind.tutorPredicate != null) {
+                    return this.tutorPredicate.equals(otherFind.tutorPredicate);
+                }
+                return false;
+            } else if (this.classPredicate != null) {
+                if (otherFind.classPredicate != null) {
+                    return this.classPredicate.equals(otherFind.classPredicate);
+                }
+                return false;
+            } else {
+                return this.keywords.equals(otherFind.keywords);
+            }
+        } else {
+            return false;
+        }
```

Comment:
can make the code shorter by just having 1 `return false` statement at the end of the function. Then the rest of the `return false` statements can be deleted; the last `else` also no need already

Reply:
oops, it's suppose to highlight the entire equals() function of FindCommand, not just those lines